### PR TITLE
[dif] autogen UART IRQ DIFs and integrate intro src tree

### PR DESCRIPTION
**Pull request by timothytrippel**
This partially addresses #8142, specifically, the item: "Integrate auto-generated IRQ DIFs into (existing) IP DIFs and deprecate manually implemented IRQ DIFs" --> "uart".

**Note**:  this PR depends on #8143 and therefore contains duplicate commits that will be removed when #8143 is merged.

### CODE REVIEW

**Comment by timothytrippel** [comment]
PR dependency removed. Ready for review.

File: sw/device/lib/dif/autogen/dif_uart_autogen.c [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+      break;
+    default:
+      return false;
+  }
+
+  return true;
+}
+
+DIF_WARN_UNUSED_RESULT
+dif_uart_result_t dif_uart_irq_get_state(
```
Comment (sriyerg) [inline sw/device/lib/dif/autogen/dif_uart_autogen.c:51]: Thanks for adding this!

**Review by sriyerg — APPROVED** [review]
LGTM, thanks!

**Comment by timothytrippel** [comment]
This was rebased on #8235 so still a dependency with some duplicate commits.

**Comment by timothytrippel** [comment]
PR dependency removed. Ready for review/merge once CI passes.

**Review by moidx — APPROVED** [review]

**Pull request merged**